### PR TITLE
STYLE: fix pylint reimported warnings

### PR DESCRIPTION
**Pull request by Moisan**
Related to https://github.com/pandas-dev/pandas/issues/48855
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_testing/contexts.py [hunk 1]
```diff
@@ -62,7 +62,6 @@ def set_timezone(tz: str) -> Generator[None, None, None]:
     ...
     'EST'
     """
-    import os
     import time
```
Comment (MarcoGorelli) [inline pandas/_testing/contexts.py:65]: `time` is in the standard library, can we move this one to the top of the file too whilst we're here?

**Review by MarcoGorelli — APPROVED** [review]
thanks @Moisan , looks good to me pending green

(related, we should probably make a check that standardlibrary imports are always at the top of the file)

**Comment by Moisan** [comment]
> (related, we should probably make a check that standardlibrary imports are always at the top of the file)

I believe the `isort` already covers this with `pyproject.toml` config:
```
sections = ["FUTURE", "STDLIB", "THIRDPARTY" ,"PRE_LIBS" , "PRE_CORE", "DTYPES", "FIRSTPARTY", "POST_CORE", "LOCALFOLDER"]
```

**Comment by MarcoGorelli** [comment]
isort just checks the order, rather than where imports are

**Pull request merged**